teps/0091-trusted-resources.md

```diff
@@ -313,12 +418,119 @@ What other approaches did you consider and why did you rule them out?  These do
+metadata:
+ name: example-tr
+spec:
+ taskRef:
+   name: example-task
+   keyref: keypath # this defines the public key of this taskref
+```
+
+```yaml
+kind: PipelineRun
+apiVersion: tekton.dev/v1beta1
+metadata:
+ name: example-pr
+spec:
+ pipelineRef:
+   name: example-pipeline
+   keyref: keypath # this defines the public key of this pipelineref
+```
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+ annotations:
+   tekton.dev/signature: MEQCIHhFC480mv7mh/6dxAlp/mvuvGXanuSghMsT+iBhWxt5AiBDvHv8sfKjJ3Ozrzvp+cjr30AOx1SPQDCcaJjlpAoVrA==
+ name: example-Pipeline
+spec:
+ tasks:
+   - name: example-task
+     taskRef:
+       name: example-task1
+       keyref: keypath1  # this defines the public key of this taskref
+   - name: example-task2
+     taskRef:
+       name: example-task2
+       keyref: keypath2  # this defines the public key of this taskref
+```
+`keyref` defines the path of the public key of this taskRef. All the tasks within a pipeline need to be signed off, and they may come from different sources with different public keys.
+So a `keyref` is needed when signing the pipeline. The signing cli doesn't need to verify the referred keys when signing the pipeline.
+
+**Pros vs Cons:**
+| Option | Pros | Cons |
+| -------- | ----------- | ----------- |
+| Config at install level (e.g. at CRD) | Easy to config at a central place | Need to loop all the keys for each verification
+| Add keyref in taskRef&pipelineRef | Each resource can specify the corresponding key, no need to loop all available keys for one resource's verification| Introduce cost for users to write and maintain YAML files
+
+### Options for integrating with remote resolution
+
+| Method | Pros | Cons |
+| -------- | ----------- | ----------- |
+| Fetch remote resources in validating Webhook | Verification fail fast in webhook | Duplicate work for Resolution, may introduce latency, and one extreme case is that the resource verified may not the the same in reconciler's resolved resource
+| Verify in Controller after resolving resources| No duplicate work for Resolution | The verification cannot fail fast in webhook. The resources may have been stored in etcd and used by other components
+| Verify in Remote Resolution | No duplicate work for Resolution | Verification coupled with Resolution
+
+In this TEP we propose to proceed with option 2 to do verification in reconciler after remote resolution is done considering the latency of doing remote resolution at admission webhook.
+
+### Options for exposing the public keys
+
+[Public keys discovery](https://transparency.dev/application/strengthen-discovery-of-encryption-keys/) is out of scope of the first phase of this TEP and considered as part of the future work. One possible mitigation is to leverage [Fulcio](https://github.com/sigstore/fulcio) to verify the identity of signing account and issue a signed certificate. The certificate is attached to the resource and can be verified later.
```

Comment:
nit: Fulcio's great for tying keys to identities, but Rekor is what you'd want to integrate with for key discovery.